code/game/machinery/pipe/construction.dm

```
@@ -24,7 +24,7 @@ Buildable meters
 /obj/item/pipe/Initialize(var/mapload, var/obj/machinery/atmospherics/P)
 	. = ..(mapload, null)
 	set_extension(src, /datum/extension/parts_stash)
-	if(!P)
+	if(!istype(P))
```

Do we want to silently ignore inappropriate instances?

---

I could throw in a `crash_with` I suppose.

---

Turns out silently failing is the only way any of this code actually works.